### PR TITLE
📦 NEW: Detailed Customer Page

### PR DESCRIPTION
**Pull request by waldorfio-dd**
# Description

- Moved search bar to header
- Display Google name and Google Profile picture in header
- Trim overflow text for long customer names in search bar
- Swizzled header/sidebar to customise DD branding
- Created a new /customers/show/:id route for each customer page
- When a customer is selected from the searchbar, redirect to new detailed customer page, and run a new query
- Detailed customer page returns all columns from the "Account" table from Salesforce, for the given customer (for now, we can trim this down to more useful information later).
- Added loading skeletons to both search bar and detailed customer page.
- Disabled test routes/ unused routes

Notes
- i have output all the columns from the Account table for now just to show that we have all the data. Will trim it down/make it more visually functional/prettier in the next update once we have discussed what information needs to stay.
- this frontend interacts with the route created in branch API-134 in the API
- this branch was created from ADP-3, which has not been merged yet (it may show more files have been changed, than have actually changed)

### Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 📦 New feature (non-breaking change which adds functionality)
- [ ] 👌 Improvement (non-breaking change which improves existing functionality)
- [ ] 🔥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📖 Documentation (non-breaking change which adds documentation)
- [ ] 📖 This change requires a documentation update in Confluence

- [ADP-4] [New]

### Checklist

- [x] Code follows the style guidelines of this project.
- [x] I have performed a self-review and testing.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where required.
- [ ] My changes generate no new warnings or errors.

### Deployment notes 💣 [x] 👍  - [ ] 👎

`yarn install`

### Screenshot or Videos 📷

https://github.com/disrupt-digital/admin-panel/assets/129021106/0cd09ee6-b2ff-47c5-a06e-badc2669f271

[ADP-4]: https://disruptdigital.atlassian.net/browse/ADP-4?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/SearchBar/CustomerSearchBar.tsx [hunk 1]
```diff
@@ -61,17 +79,41 @@ export function CustomerSearchBar () {
...
+      )}
       onInputChange={handleInputChange}
       renderOption={(props, option) =>
-        <li {...props} key={option.key}>
-          {option.label} {option.brand ? `(${option.brand})` : ''}
+        <li
+          {...props}
+          key={option.key}
+          onClick={() => handleClick(option.value)}
+
```
Comment (andrewk-dd) [inline src/components/SearchBar/CustomerSearchBar.tsx:103]: This empty line can be removed

File: src/components/layout/sider.tsx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+            </div>
+          </CanAccess>
+        )
+      }
+
+      const linkStyle: CSSProperties = activeItemDisabled && isSelected ? { pointerEvents: 'none' } : {}
+
+      return (
+        <CanAccess key={item.key} resource={name.toLowerCase()} action="list" params={{ resource: item }}>
+          <Tooltip title={label ?? name} placement="right" disableHoverListener={!siderCollapsed} arrow>
```
Comment (andrewk-dd) [inline src/components/layout/sider.tsx:151]: Is there a way to split this file to multiple independent components? It's a bit hard to understand what is going on there

**Review by andrewk-dd — APPROVED** [review]

**Comment by waldorfio-dd** [comment]
@andrewk-dd ready for re-review

File: src/components/layout/sider.tsx [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+import Tooltip from '@mui/material/Tooltip'
+import type { RefineThemedLayoutV2SiderProps } from '@refinedev/mui'
+
+export const ThemedSiderV2: React.FC<RefineThemedLayoutV2SiderProps> = ({ Title: TitleFromProps, render, meta, activeItemDisabled = false }) => {
+  const { siderCollapsed, setSiderCollapsed, mobileSiderOpen, setMobileSiderOpen } = useThemedLayoutContext()
+
+  const drawerWidth = () => {
+    if (siderCollapsed) return 56
+    return 240
+  }
```
Comment (andrewk-dd) [inline src/components/layout/sider.tsx:32]: Not an issue, just an optional suggestion
```suggestion
  const drawerWidth = () => siderCollapsed ? 56 : 240
```

File: src/components/layout/sider.tsx [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+
+  const t = useTranslate()
+  const routerType = useRouterType()
+  const Link = useLink()
+  const { Link: LegacyLink } = useRouterContext()
+  const ActiveLink = routerType === 'legacy' ? LegacyLink : Link
+  const { hasDashboard } = useRefineContext()
+  const translate = useTranslate()
```
Comment (andrewk-dd) [inline src/components/layout/sider.tsx:40]: Should you remove `t` in this case?

File: src/components/layout/sider.tsx [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+  const { menuItems, selectedKey, defaultOpenKeys } = useMenu({ meta })
+  const isExistAuthentication = useIsExistAuthentication()
+  const TitleFromContext = useTitle()
+  const authProvider = useActiveAuthProvider()
+  const { warnWhen, setWarnWhen } = useWarnAboutChange()
+  const { mutate: mutateLogout } = useLogout({
+    v3LegacyAuthProviderCompatible: Boolean(authProvider?.isLegacy),
+  })
+
+  const [open, setOpen] = useState<{ [k: string]: boolean }>({})
```
Comment (andrewk-dd) [inline src/components/layout/sider.tsx:51]:
```suggestion
  const [open, setOpen] = useState<Record<string, boolean>>({})
```

File: src/components/layout/sider.tsx [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+  const RenderToTitle = TitleFromProps ?? TitleFromContext ?? DefaultTitle
+
+  const handleClick = (key: string) => {
+    setOpen({ ...open, [key]: !open[key] })
+  }
+
+  const renderTreeView = (tree: ITreeMenu[], innerSelectedKey?: string) => {
+    return tree.map((item: ITreeMenu) => {
+      const { icon, label, route, name, children, parentName, options } = item
+      const { innerMeta } = item.meta;
```
Comment (andrewk-dd) [inline src/components/layout/sider.tsx:73]:
```suggestion
      const { innerMeta } = item.meta
```

File: src/components/layout/sider.tsx [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+              {!siderCollapsed && (
+                <Collapse in={open[item.key || '']} timeout="auto" unmountOnExit>
+                  <List component="div" disablePadding>
+                    {renderTreeView(children, selectedKey)}
+                  </List>
+                </Collapse>
+              )}
+            </div>
+          </CanAccess>
+        )
```
Comment (andrewk-dd) [inline src/components/layout/sider.tsx:147]:
```suggestion
        ).filter(Boolean)
```
as some values of the array can be undefined (as the result of this condition `if (children.length > 0) {`)

**Review by andrewk-dd — APPROVED** [review]

**Comment by waldorfio-dd** [comment]
@andrewk-dd

File: src/components/layout/sider.tsx (hunk 4 above)
Comment (andrewk-dd) [reply src/components/layout/sider.tsx:37]: Any response?

**Review by andrewk-dd — APPROVED** [review]

**Pull request merged**